transports/noise/src/io/handshake.rs

```diff
@@ -207,19 +207,25 @@ where
     }
     let pb = pb_result?;
 
-    if !pb.identity_key.is_empty() {
-        let pk = identity::PublicKey::from_protobuf_encoding(&pb.identity_key)
-            .map_err(|_| NoiseError::InvalidKey)?;
-        if let Some(ref k) = state.id_remote_pubkey {
-            if k != &pk {
-                return Err(NoiseError::InvalidKey);
+    match pb.identity_key {
+        Some(ref identity_key) => {
+            let pk = identity::PublicKey::from_protobuf_encoding(&identity_key)
+                .map_err(|_| NoiseError::InvalidKey)?;
+            if let Some(ref k) = state.id_remote_pubkey {
+                if k != &pk {
+                    return Err(NoiseError::InvalidKey);
+                }
             }
+            state.id_remote_pubkey = Some(pk);
         }
-        state.id_remote_pubkey = Some(pk);
+        None => {}
```

Comment:
@thomaseizinger Thanks for the tips on panic/error checking/match, that was a a good bit to read up on as I lean rust. I think this code both fulfills the spirit of what was happening before (if there was no key, just keep going) but now does it without the panic (I think).

Again, thank you for your guidance on this. I really appreciate it.

Please let me know if I'm moving in the right direction, or if any other changes are needed.

Reply:
If you run `clippy`, it will tell you that you can replace this with an `if let Some(identity_key)` construct :)